wof/WaterML_1_1.py

```diff
@@ -402,7 +402,7 @@ def quote_xml(inStr):
         pos = mo.end()
     s3 = s1[pos:]
     s2 += quote_xml_aux(s3)
-    return s2.decode(ExternalEncoding)
+    return s2.encode(ExternalEncoding).decode(ExternalEncoding)
```

Comment:
@lsetiawan the `ExternalEncoding` is `utf-8`, correct?

We are playing the dangerous game of encode/decode whack-a-mole :smile: 

The `WaterML_1_1.py` module is gigantic, >7500 lines :scream:  Any operation here is non-trivial and we are barely capable of doing damage control.

The usual logic that we should try is:

- read everything as bytes (xml, etc), if it is not already bytes we should encode it;
- use bytes internally in the API;
- decode only to write.

I must confess that I have no idea what we are reading and writing here. Maybe we need a live call to discuss a long term goal for this, specially if Python 3 is you ODM2 sights.

Reply:
> @lsetiawan the ExternalEncoding is utf-8, correct?

Yes it is.

Reply:
> I must confess that I have no idea what we are reading and writing here.

We are reading values from a database and writing Water ML XML representation. The failures comes when trying to writing that into XML.

Reply:
I am trying something here... Let's see... Hold on!

Reply:
@ocefpaf: seriously, there's no rush on this!! Go back to folium to finish it off and ship the release! Unless you enjoy distractions :stuck_out_tongue_winking_eye: 

We can reconnect next week, to discuss both these unicode issues and longer term goals.